JenkinsFile

```diff
@@ -0,0 +1,129 @@
+pipeline {
+    agent {
+        dockerfile {
+            label 'linux'
+        }
+    }
+    environment {
+        GIT_CREDENTIALS = credentials("${GITHUB_CREDENTIALS}")
+    }
+    stages {
+        stage('Lint/Test') {
+            steps {
+                sh """
+                    npm i
+                    npm run lint
+                    npm run test
+                """
+            }
+        }
+
+        stage('Version Check') {
+            steps {
+                 script {
+                    def data = readJSON file:'package.json'
+                    checkVersionAlreadyExists(data.version)
+                }
+            }
+        }
+
+        stage('Publish') {
+            steps {
+                sh """
+                    npm run build
+                    npm set registry "https://registry.npmjs.org"
+                    npm set //registry.npmjs.org/:_authToken ${NPM_TOKEN}
+                    node build_release_package_json.js
+                    node create_release_readme.js
+                    npm publish ./dist
+                """
+            }
+        }
+
+        stage('Tag') {
+            steps {
+                script {
+                    def data = readJSON file:'package.json'
+                    tagRepo(data.version)
+                }
+            }
+        }
+    }
+    post {
+        success {
+            notifyComplete('SUCCESS')
+        }
+        failure {
+            notifyComplete('FAILURE')
```

Comment:
Let's copy the logic from RevAiApi and RevCommon for not showing failure messages if the previous build was a failure. This will reduce bot spam in our development slack channel.